File: java/dbos-starter/src/main/java/com/example/dbos_starter/config/DurableStarterConfig.java
package com.example.dbos_starter.config;

import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.context.annotation.Primary;

import com.example.dbos_starter.service.DurableStarterService;
import com.example.dbos_starter.service.DurableStarterServiceImpl;

import dev.dbos.transact.DBOS;

@Configuration
public class DurableStarterConfig {

    @Primary
    @Bean
    public DurableStarterService durableStarterService() {
	    var proxy = DBOS.registerWorkflows(
            DurableStarterService.class,
            new DurableStarterServiceImpl());
        return proxy;
    }
}

File: kotlin/dbos-starter/src/main/kotlin/com/example/dbos_starter/config/DurableStarterConfig.kt
package com.example.dbos_starter.config

import com.example.dbos_starter.service.DurableStarterService
import com.example.dbos_starter.service.DurableStarterServiceImpl
import dev.dbos.transact.DBOS
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.context.annotation.Primary
import registerWorkflows

@Configuration
class DurableStarterConfig {

    @Primary
    @Bean
    fun durableStarterService(): DurableStarterService {
        return registerWorkflows<DurableStarterService>(DurableStarterServiceImpl())
    }
}